lib/core/hooks.js

```diff
@@ -34,24 +37,14 @@ module.exports = function (env_config, callback) {
     return false;
   };
 
-  // todo: refactor run() to use this
-  var getHooksFromPlugins = function(type, name) {
-    return _.compact(_.map(env_config.plugins, function(plugin) {
-      if(plugin.hooks[type] && plugin.hooks[type][name]) {
-        return plugin.hooks[type][name];
-      }
-    }));
-  };
-
   var runAsyncEvery = function(type, name, args, callback) {
-    var runHook = function(hook, callback) {
-      args.push(callback);
```

Comment:
Every time `runHook` is called a new `callback` function is pushed to the `args` array. I don't think it should work this way.